src/app/modules/category/components/filter/filter.component.ts

```diff
@@ -1,12 +1,202 @@
+                            filterItem.first_filter_key,
+                    );
+                    const indexItem = currentData.indexOf(
+                        itemFromFilteredArray,
+                    );
+                    currentData.splice(indexItem, 1);
+                }
+                this.shareService.sendData(currentData);
+            });
+    }
+
+    onInputFirst(id: number, value: FiltersData): void {
+        this.filterItemFromTo = {
+            first_filter_key: value,
+            id,
+        };
+        this.onFilterFromTo(this.filterItemFromTo);
+    }
+
+    onInputSecond(id: number, value: FiltersData): void {
+        this.filterItemFromTo = {
+            ...this.filterItemFromTo,
+            second_filter_key: value,
+            id,
+        };
+        this.onFilterFromTo(this.filterItemFromTo);
+    }
+
+    onSelectFirst(id: number, value: FiltersData): void {
+        this.filterItemFromTo = {
+            ...this.filterItemFromTo,
+            first_filter_key: value,
+            id,
+        };
+        this.onFilterFromTo(this.filterItemFromTo);
+    }
+
+    onSelectSecond(id: number, value: FiltersData): void {
+        this.filterItemFromTo = {
+            ...this.filterItemFromTo,
+            second_filter_key: value,
+            id,
+        };
+        this.onFilterFromTo(this.filterItemFromTo);
+    }
+
+    onFilterFromTo(filterFromTo: FilterItem): void {
+        this.selectedItemsArray$
+            .pipe(take(1))
+            .subscribe((data: FilterItem[]) => {
+                const res = data.find(el => el.id === filterFromTo.id);
+                if (res) {
+                    (res.first_filter_key = filterFromTo.first_filter_key),
+                        (res.second_filter_key =
+                            filterFromTo.second_filter_key);
+                } else {
+                    data.push(filterFromTo);
+                }
+                this.removeFromTo();
+                this.shareService.sendData(data);
+            });
+    }
+
+    removeFromTo(): void {
+        this.selectedItemsArray$
+            .pipe(take(1))
+            .subscribe((data: FilterItem[]) => {
+                if (data === null) {
+                    data = [];
+                }
+                const filteredFrom = data.find(
+                    item =>
+                        item.first_filter_key === '' ||
+                        item.first_filter_key === undefined,
+                );
+                const filteredTo = data.find(
+                    item =>
+                        item.second_filter_key === '' ||
+                        item.second_filter_key === undefined,
+                );
+
+                const filteredIndexFrom = data.indexOf(filteredFrom);
+                const filteredIndexTo = data.indexOf(filteredTo);
+
+                if (filteredIndexFrom > -1 && filteredIndexTo > -1) {
+                    data.splice(filteredIndexFrom, 1);
```

Comment:
here you change inner observable value through splice function, don't do that
better to send new array to the service